src/js/hooks/cycleCount/useAllProductsTab.jsx

```diff
@@ -0,0 +1,176 @@
+        </TableHeaderCell>
+      ),
+      cell: ({ getValue }) => (
+        <TableCell className="rt-td">
+          {getValue()}
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor((row) => `${row.product.productCode} ${row.product.name}`, {
+      id: 'product',
+      header: () => (
+        <TableHeaderCell sortable>
+          {translate('react.cycleCount.table.products.label', 'Products')}
+        </TableHeaderCell>
+      ),
+      cell: ({ getValue, row }) => (
+        <TableCell
+          link={INVENTORY_ITEM_URL.showStockCard(row.original.product.productCode)}
+          className="rt-td multiline-cell"
+        >
+          {getValue()}
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor('product.category', {
+      header: () => (
+        <TableHeaderCell sortable>
+          {translate('react.cycleCount.table.category.label', 'Category')}
+        </TableHeaderCell>
+      ),
+      cell: ({ getValue }) => (
+        <TableCell className="rt-td multiline-cell">
+          {getValue()}
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor('binLocation.name', {
+      header: () => (
+        <TableHeaderCell sortable>
+          {translate('react.cycleCount.table.binLocation.label', 'Bin Location')}
+        </TableHeaderCell>
+      ),
+      cell: ({ getValue }) => (
+        <TableCell
+          className="rt-td"
+          tooltip
+          tooltipLabel={getValue()}
+        >
+          {getValue()}
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor('product.tags', {
+      header: () => (
+        <TableHeaderCell>
+          {translate('react.cycleCount.table.tag.label', 'Tag')}
+        </TableHeaderCell>
+      ),
+      cell: ({ getValue }) => (
+        <TableCell className="rt-td multiline-cell">
+          {getValue()}
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor('product.catalogs', {
+      header: () => (
+        <TableHeaderCell>
+          {translate('react.cycleCount.table.productCatalogue.label', 'Product Catalogue')}
+        </TableHeaderCell>
+      ),
+      cell: ({ getValue }) => (
+        <TableCell className="rt-td multiline-cell">
+          {getValue()}
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor('product.abcClass', {
+      header: () => (
+        <TableHeaderCell sortable>
+          {translate('react.cycleCount.table.abcClass.label', 'ABC Class')}
+        </TableHeaderCell>
+      ),
+      cell: ({ getValue }) => (
+        <TableCell className="rt-td">
+          {getValue()}
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor('product.quantity', {
+      header: () => (
+        <TableHeaderCell sortable>
+          {translate('react.cycleCount.table.quantity.label', 'Quantity')}
+        </TableHeaderCell>
+      ),
+      cell: ({ getValue }) => (
+        <TableCell className="rt-td">
+          {getValue()}
+        </TableCell>
+      ),
+    }),
+  ], []);
```

Comment:
I can add it, but I am not sure how it will work, because translate is a function, so I don't think it will be working properly, it should rather be a selected language

Reply:
good point, I forgot that `translate` is a function. I think it's good to investigate places where we put it in the dependency, as we might have some unexpected rerenders there. And yeah, selected language is the option here.